src/Sulu/Bundle/TestBundle/Testing/TestUserProvider.php

```diff
@@ -87,21 +97,46 @@ public function getUser()
             $this->entityManager->persist($contact);
 
             $user = $this->userRepository->createNew();
+            $user->setUsername($username);
             $user->setContact($contact);
             $this->entityManager->persist($user);
         }
 
-        $user->setUsername(self::TEST_USER_USERNAME);
         $user->setSalt('');
         $encoder = $this->userPasswordEncoderFactory->getEncoder($user);
-        $user->setPassword($encoder->encodePassword('test', $user->getSalt()));
+        $user->setPassword($encoder->encodePassword($username, $user->getSalt()));
         $user->setLocale('en');
 
         $this->entityManager->flush();
 
-        $this->user = $user;
+        if (self::TEST_USER_USERNAME === $username) {
+            $this->user = $user;
+        }
+
+        return $user;
+    }
+
+    public function getRole(
```

Comment:
As above, I would not include it in this class, because it is not really its responsibility to create roles. The implementation is also a bit confusing IMO, because it's not really a getter nor a creator. Calling it somehow like would cause the second line to change the role of the first line as well:

```php
$role1 = $testUserProvider->getRole('Test', 'sulu', false);
$role2 = $testUserProvider->getRole('Test', 'website', true);
```